container/new_push.bzl

```diff
@@ -80,16 +77,37 @@ def _impl(ctx):
         if len(ctx.files.image) > 1:
             fail("Attribute image {} to {} had {} files. Expected exactly 1".format(ctx.attr.image, ctx.label, len(ctx.files.image)))
         pusher_args += ["-src", _get_runfile_path(ctx, ctx.files.image[0])]
+        digester_args += ["-src", ctx.files.image[0].path]
+        digester_input = [ctx.files.image[0]]
     if ctx.attr.format == "legacy":
-        legacy_dir = generate_legacy_dir(ctx)
+        # Construct container_parts for input to pusher.
+        image = _get_layers(ctx, ctx.label.name, ctx.attr.image)
+        legacy_dir = generate_legacy_dir(ctx, image["config"], image["manifest"], image.get("zipped_layer", []))
         temp_files, config = legacy_dir["temp_files"], legacy_dir["config"]
 
         pusher_args += ["-src", "{}".format(_get_runfile_path(ctx, config))]
+        digester_args += ["-src", str(config.path)]
+        digester_input = temp_files
 
-        for layer_path in legacy_dir["layers"]:
-            pusher_args += ["-layers", "{}".format(_get_runfile_path(ctx, layer_path))]
+        for layer_file in legacy_dir["layers"]:
+            pusher_args += ["-layers", "{}".format(_get_runfile_path(ctx, layer_file))]
+            digester_args += ["-layers", layer_file.path]
 
-    pusher_args += ["-format", str(ctx.attr.format)]
+    pusher_args += ["-dst", "{registry}/{repository}:{tag}".format(
+        registry = registry,
+        repository = repository,
+        tag = tag,
+    ), "-format", str(ctx.attr.format)]
```

Comment:
nit: use another `pusher_args += ` for `-format`. This way is slightly unreadable.